packages/backend/src/services/ServiceRepository.ts

```diff
@@ -94,111 +159,222 @@ abstract class ServiceRepositoryBase {
  * NOTE: For now, this repository simply exposes services instantiated in `./services.ts`,
  *       and provided to this repository directly. At a later stage, this repository will
  *       handle instantiating all services internally, including cross-service dependencies.
- */ export class ServiceRepository
+ */
+export class ServiceRepository
     extends ServiceRepositoryBase
     implements ServiceFactoryMethod<ServiceManifest>
 {
+    /**
+     * Holds memoized instances of services after their initial instantiation:
+     */
+    protected _serviceCache: Partial<ServiceManifest> = {};
```

Comment:
If this looks like it does the same thing as `_services`: it does! One of them will be removed once we're no longer sharing services outside of the repository.

Reply:
That means once we remove `services` the methods would be like this?

```
    public getAnalyticsService(): AnalyticsService {
        return this.getService(
            'analyticsService',
            () => new AnalyticsService(...),
        );
    }
```

Reply:
Correct!